File: qme/static/types/resume.ts
export interface PersonalInfo {
  fullName: string;
  email: string;
  phone: string;
  location: string;
  linkedinUrl: string;
}

export interface WorkExperience {
  id: string;
  company: string;
  position: string;
  startDate: string;
  endDate: string;
  description: string;
  current: boolean;
}

export interface Education {
  id: string;
  school: string;
  degree: string;
  field: string;
  graduationDate: string;
}

export interface Skill {
  id: string;
  name: string;
  level?: 'beginner' | 'intermediate' | 'advanced';
}

export interface ResumeData {
  personalInfo: PersonalInfo;
  summary: string;
  workExperience: WorkExperience[];
  education: Education[];
  skills: Skill[];
  certificates?: string[];
}

export interface ResumeStyle {
  id: string;
  name: string;
  description: string;
  cssClass: string;
}

File: qme/schemas/resume.py
from pydantic import BaseModel, EmailStr

class PersonalInfo(BaseModel):
    name: str
    email: EmailStr
    phone: str

class Education(BaseModel):
    institution: str
    degree: str
    start_year: int
    end_year: int

class Experience(BaseModel):
    company: str
    title: str
    start_date: str  # Use ISO format: YYYY-MM-DD
    end_date: str  # Use ISO format: YYYY-MM-DD
    description: str

class Skill(BaseModel):
    name: str
    proficiency: str  # e.g., 'beginner', 'intermediate', 'expert'

class Resume(BaseModel):
    personal_info: PersonalInfo
    education: list[Education]
    experience: list[Experience]
    skills: list[Skill]  
